hepcrawl/extractors/iop_parser.py

```diff
@@ -124,3 +122,32 @@ def get_arxiv_eprints(self, node):
             arxiv_eprints.append({'value': pattern.sub('', arxiv_value)})
 
         return arxiv_eprints
+
+    def get_license_type_and_version_from_url(self, url):
+        match = self.LICENSE_PATTERN.search(url)
+        if "creativecommons.org/licenses/by/" in url:
+            license_type = "CC-BY"
+            version = match.group(2)
+            return [self.construct_license(license_type=license_type, version=version, url=url)]
+        else:
+            raise UnknownLicense(url)
+
+    def _get_license(self, node):
+        licenses = node.xpath("//permissions/license/@href").extract()
+        for license in licenses:
+            try:
+                type_and_version = self.get_license_type_and_version_from_url(url=license)
+                if type_and_version:
+                    return type_and_version
+            except KeyError:
+                    logger.error("License is not found in XML.")
+        return licenses
+
+
+    def construct_license(self, license_type, version, url=None):
+        if url and license_type and version:
+            return {"url": url, "license": "-".join([license_type, version])}
+        if license_type and version:
+            logger.warning("License URL is not found in XML.")
+            return {"license": "-".join([license_type, version])}
```

Comment:
```suggestion
        license = {}
        if  license_type and version:
            license['license'] = "-".join([license_type, version])
        if url:
            license['url'] = url
        return license
```

Reply:
I would suggest keeping the warning for license URLs, since for displaying the license in the right corner of the article, we need to have license and url values:
<img width="1302" alt="Screenshot 2023-09-26 at 15 37 21" src="https://github.com/SCOAP3/hepcrawl/assets/36658177/a652a5d6-a53c-4e76-91ca-3c14dc013b9c">